ci_framework/roles/use_install_yamls/library/create_env_file.py

```diff
@@ -0,0 +1,176 @@
+
+# Return install_yamls vars
+def _get_makefile_vars(makefile_path):
+    if _check_makefile(makefile_path):
+        with open(makefile_path) as f:
+            content = f.read().split('\n')
+    # Get a list of vars
+    vars_list = [data.split('?=') for data in content if '?=' in data]
+    vars = {k.strip().lower(): v.strip().lower() for k,v in vars_list}
+    return vars        
+
+# Check if vars exists in install_yamls or not
+def _check_vars(makefile_path, users_var):
+    var_not_found = []
+    makefile_vars = _get_makefile_vars(makefile_path)
+    for key in users_var.keys():
+        if key not in makefile_vars.keys():
+            var_not_found.append(key)
+
+    return var_not_found
+
+# Create environment file
+def _create_file(envfile_name, users_var, output_dir):
+    env_path = os.path.join(output_dir, envfile_name)
+    with open(env_path, 'w') as f:
+        for k,v in users_var.items():
+            f.write(f"{k.upper()}={v}\n")
+    # Check whether file exists or not
+    if os.path.exists(env_path):
+        return True
+    else:
+        False
+
+# Create yaml file
+def _create_yaml_file(envfile, users_var, output_dir):
+    yaml_path = os.path.join(output_dir, envfile_name, '.yaml')
+    users_var_dict = {k.upper():v for k,v in users_var.items()}
+    with open(yaml_path, 'w') as file:
+        yaml.dump(users_var_dict,file)    
+
+# Run the module
+def create_env_file(envfile, makefile, check_var, users_var, output_dir, create_yaml=False):
+    unwanted_vars = []
+    if check_var:
+        unwanted_vars = _check_vars(makefile, users_var)
+    if not unwanted_vars:
+        if create_yaml:
+            _create_yaml_file(envfile, users_var, output_dir)
+        return _create_file(envfile, users_var, output_dir)
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict(
+        output_directory=dict(type='str', required=True),
+        makefile_path=dict(type='str', required=True),
+        install_yamls_var=dict(type='dict', required=True),
+        envfile_name=dict(type='str', default='install_yamls'),
+        check_var=dict(type='bool', default=True),
+        create_yaml=dict(type='bool', default=False),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=False
+    )
+    
+    result = dict(
+        success=False,
+        changed=False,
+        error="",
+    )
+    
+    # Get the module output
+    output_dir = module.params.get("output_directory")
+    envfile = module.params.get("envfile_name")
+    makefile = module.params.get("makefile_path")
+    check_var = module.params.get("check_var")
+    users_var = module.params.get("install_yamls_var")
+    create_yaml = module.params.get("create_yaml")
+
+    status = create_env_file(envfile, makefile, check_var, users_var, output_dir, create_yaml)
+
+    if status:
+        result["Success"] = True
+    else:
+        result["msg"] = status
```

Comment:
the status here would be either None, True or False, no meaning message if success is False